src/config.js

```diff
@@ -208,6 +235,25 @@ export const rtc = Object.freeze({
         maxSctpMessageSize: MAX_BUF_IN,
         sctpSendBufferSize: MAX_BUF_OUT,
     },
+    producerOptionsByKind: {
+        /** @type {import("mediasoup-client").types.ProducerOptions} */
+        audio: baseProducerOptions,
+        /** @type {import("mediasoup-client").types.ProducerOptions} */
+        video: {
+            ...baseProducerOptions,
+            // for browsers using libwebrtc, values are set to allow simulcast layers to be made in that range
+            codecOptions: {
+                videoGoogleMinBitrate: 1_000,
+                videoGoogleStartBitrate: 1_000_000,
+                videoGoogleMaxBitrate: MAX_VIDEO_BITRATE * 2,
+            },
+            encodings: [
+                { scaleResolutionDownBy: 4, maxBitrate: Math.floor(MAX_VIDEO_BITRATE / 4) },
+                { scaleResolutionDownBy: 2, maxBitrate: Math.floor(MAX_VIDEO_BITRATE / 2) },
+                { scaleResolutionDownBy: 1, maxBitrate: MAX_VIDEO_BITRATE },
+            ],
```

Comment:
Maybe I'm wrong, but a scale down of `4` implies a reduction of bitrate by `16`, not `4`, because it's a fourth of x and a fourth of y, resulting to 16 less pixels to render for an equivalent framerate, thus roughtly a bitrate divided by 16.

Of course, this math works with the bitrate scaling with pixels, which is not necessarily entirely true with the compression algorithms at hand. Probably the scale down has diminishing return of reduced bitrate if you take the compression algorithm. but probably not to the point it's close to linear with `1080p` for example.

Reply:
Yes the resolution should scale down by the square of the value, but as you mentioned it may not scale exactly like that and there is likely some constant overhead on top of it, if we are too strict we may lose packets at the lower resolutions. Since it's a cap it shouldn't be a problem to have a generous margin since the lower resolutions will only use the bitrate they need.